Feature: Mesh Generalization

tmr9209 opened the pull request:
Assimp mesh loading is implemented.

tmr9209 committed:
Generalized mesh into fbxmesh and objmesh classes.

tmr9209 committed:
Fixed error, compiles now.

tmr9209 committed:
Merge branch 'develop' into feature/MeshGeneralization

tmr9209 committed:
Merge branch 'develop' into feature/MeshGeneralization

Fixed issues with merging in mesh.d

Conflicts:
	source/components/mesh.d

tmr9209 committed:
Reimplemented objmesh and fbxmesh back in to a working state.

Commit:
Added a few changes to objmesh and prepared for implementing Assimp.

tmr9209 committed:
Added assimp into the project (dll, package.json, bindings).

tmr9209 committed:
Implementing Assimp.  Currently assimp loads data, still need to parse assimp data into vertices and indices.

tmr9209 committed:
Implemented loading meshes with assimp library.  Removed unnecessary objmesh and fbxmesh. (objects can now be loaded in any format with any vertex data)

wozniakty commented:
:+1:

wozniakty commented:
For everyone else: this pull request is for the initial version of this feature so that rendering can make use of it.  Tim plans to clean it up and add some more features by the end of the milestone, so expect a second pull request this weekend.

tmr9209 committed:
Added missing assimp32.dll.

brooklynlittell commented:
The object I now have drawing looks nothing like what is has previously. Is this correct for this initial PR?

dislogical commented on source/components/mesh.d:
```diff
@@ -27,91 +28,67 @@ public:
-				formattedRead( line, "vn %s %s %s",
-							   &x, &y, &z );
+			// Did not load
+			Output.printValue( OutputType.Info, "Error", "Mesh not loaded." );
```
This should probably be:

```
Output.printMessage( OutputType.Error, "Mesh not loaded." );
```

But it can wait until V2.

dislogical commented:
:a:

tmr9209 committed:
Figured out how to create a new and update assimp32.dll using cmake, added it in.
Added a test fbx model with diffuse/normal map. (Must be only model in Meshes folder to see)
Fixed debug error message colden wanted fixed.

wozniakty commented:
:+1:

tmr9209 committed:
Fixed error message.
